app/common/TimeQuery.ts

```diff
@@ -0,0 +1,234 @@
+import {ActionSummary, ColumnDelta, createEmptyActionSummary, createEmptyTableDelta} from 'app/common/ActionSummary';
+import {CellDelta} from 'app/common/TabularDiff';
+import {concatenateSummaries} from 'app/common/ActionSummarizer';
+import cloneDeep = require('lodash/cloneDeep');
+import keyBy = require('lodash/keyBy');
+import matches = require('lodash/matches');
+import sortBy = require('lodash/sortBy');
+import toPairs = require('lodash/toPairs');
+
+/**
+ * We can combine an ActionSummary with the current state of the database
+ * to answer questions about the state of the database in the past.  This
+ * is particularly useful for grist metadata tables, which are needed to
```

Comment:
```suggestion
 * is particularly useful for Grist metadata tables, which are needed to
```

In most places we use uppercase latter :)